Detectors/TOF/simulation/include/TOFSimulation/Digitizer.h

```diff
@@ -124,6 +129,8 @@ class Digitizer
 
   o2::dataformats::MCTruthContainer<o2::tof::MCLabel> mFutureMCTruthContainer;
 
+  CalibApi* mCalibApi = nullptr; //! calib api to handle the TOF calibration
```

Comment:
Does the digitizer itself need to know about CCDB access? The alternative would be to pass-in the calibration objects instead of letting the digitizer look them up. Like this you could pass in any object, coming from CCDB or not.

Reply:
Digitizer doesn't know about CCDB. CalibApi is our interface which manages calibration objects independently of they come from ccdb or are passed by somebody else.

In the current tof workflow we pass objects from ccdb or default objects to calibapi depending on the parameters given to dpl.

The current configuration of our workflow doesn't require ccdb access.

Does this answer your questions?

Reply:
Partly. I may be misinterpreting the code. My question is: Who needs to take action to get a new object and when? Is it the digitizer itself? In this case we put it into an active role.  Alternatively it be some entity outside (DPL processor) so that the digitizer does not need to do the query but merely use what is given it.

Reply:
At the moment the players which can set a new object are the specs which are supposed to load calibrations: TOFDigitizerSpec (for decalibration) and TOFClusterizerSpec (for calibration).

We planned to manage the CalibApi inside the run processes of those specs.
Currently, we didn't implement yet an intelligent way to change it since we still don't have a real access to ccdb.
At some points we should manage also this feature but the case it is automatically done at the level of the ccdb service spec of Giulio.

If you have different suggestions let us know. In any case we would prefer to have a single point (CalibApi) where to update objects and recall it in all the pieces which need calibrations (currently digitizer and clusterizer objects).

Reply:
For completeness ,
if you run 
o2-testworkflows-tof-dummy-ccdb |o2-sim-digitizer-workflow
o2-testworkflows-tof-dummy-ccdb |o2-tof-reco-workflow

without any extra flag, access to ccdb is not requested and workflows work as before.